src/features/Instructors/InstructorsFilters/index.jsx

```diff
@@ -1,73 +1,142 @@
-import React, { useState } from 'react';
+import React, {
+  useState, useReducer, useContext, useEffect,
+} from 'react';
 import {
-  DropdownButton, Form, Col, Button,
+  Form, Col,
 } from '@edx/paragon';
+import { camelCaseObject } from '@edx/frontend-platform';
+import { Select, Button } from 'react-paragon-topaz';
+import { logError } from '@edx/frontend-platform/logging';
+import { InstitutionContext } from 'features/Main/institutionContext';
+import reducer from 'features/Instructors/InstructorsFilters/reducer';
+import { getCoursesByInstitution } from 'features/Common/data/api';
+import {
+  FETCH_COURSES_DATA_REQUEST,
+  FETCH_COURSES_DATA_SUCCESS,
+  FETCH_COURSES_DATA_FAILURE,
+} from 'features/Instructors/actionTypes';
+import { RequestStatus } from 'features/constants';
 import PropTypes from 'prop-types';
 
-const initialFilterFormValues = {
-  instructorName: '',
-  instructorEmail: '',
-  ccxId: '',
+const initialState = {
+  courses: {
+    data: [],
+    status: RequestStatus.SUCCESS,
+    error: null,
+  },
 };
 
 const InstructorsFilters = ({ fetchData, resetPagination }) => {
-  const [filters, setFilters] = useState(initialFilterFormValues);
+  const stateInstitution = useContext(InstitutionContext);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [courseOptions, setCourseOptions] = useState([]);
+  const [instructorName, setInstructorName] = useState('');
+  const [instructorEmail, setInstructorEmail] = useState('');
+  const [courseSelected, setCourseSelected] = useState(null);
+  // check this after implementation of selector institution
+  let id = 0;
+  if (stateInstitution.length > 0) {
```

Comment:
What about the case `stateInstitution.length` is equal to 1? Thats the case whe the user is an institution admin, then the endpoint of courses is not called in that case:

![Screenshot from 2023-12-05 19-21-36](https://github.com/Pearson-Advance/frontend-app-institution-portal/assets/30726391/7d186d71-7103-450e-885e-473b45e52ce4)